### PR TITLE
style: create base button component

### PR DESCRIPTION
**Pull request by dgd03146**
Created reusable button component with tw-merge and clsx library.

### CODE REVIEW

File: components/UI/ListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    DetailedHTMLProps<LiHTMLAttributes<HTMLLIElement>, HTMLLIElement>
+  > {
+  right?: React.ReactNode;
+}
+
+const ListItem = ({ right, children, ...props }: Props) => {
+  return (
+    <li {...props}>
+      {children}
+      {right}
```
Comment (iammminzzy) [inline components/UI/ListItem.tsx:18]: What does "right" mean? Isn't that included in `children`?

File: components/UI/List.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ReactNode,
+} from 'react';
+
+import { cn } from '@/utils/cn';
+
+import ListItem from './ListItem';
+
+interface ListProps<T>
+  extends PropsWithChildren<
+    DetailedHTMLProps<HTMLAttributes<HTMLUListElement>, HTMLUListElement>
```
Comment (iammminzzy) [inline components/UI/List.tsx:14]:
```suggestion
  extends PropsWithChildren<HTMLUListElement>
```
 is not working?

File: components/UI/ListItem.tsx (hunk 1 above)
Comment (dgd03146) [reply components/UI/ListItem.tsx:18]: right means literally component will be placed in right side

File: components/UI/List.tsx (hunk 2 above)
Comment (dgd03146) [reply components/UI/List.tsx:14]: I think those are okay.
image
https://stackoverflow.com/questions/58418950/whats-the-difference-between-detailedhtmlprops-and-htmlattributes

**Review by iammminzzy — APPROVED** [review]

**Pull request merged**